app/services/submit_application.rb

```diff
@@ -18,7 +18,12 @@ def call
       submit_application
     end
 
-    CandidateMailer.application_submitted(application_form).deliver_later
+    if send_to_provider_immediately?
+      CandidateMailer.application_sent_to_provider(@application_form).deliver_later
+    else
+      CandidateMailer.application_submitted(application_form).deliver_later
+    end
```

Comment:
I guess these could cause a kind of race condition if the mailer depends on something happening in the current transaction. But that's not a new problem. I think I'd be in favour of adding a `deliver_later(wait: 5.seconds)` or something if this is an issue.

Reply:
Hmm, not aware of that being an issue currently. Fine to merge without?